### PR TITLE
Code changes from netflix main to ciscom31

### PR DESCRIPTION
**Pull request by fthankac**

### CODE REVIEW

**Review by saralaks — COMMENTED** [review]
Migrate 'pruneWorkflows' and 'pruneTasks' methods from ES5 to ES7.

**Comment by fthankac** [comment]
> Migrate 'pruneWorkflows' and 'pruneTasks' methods from ES5 to ES7.

Done migrating 'pruneWorkflows' and 'pruneTasks' methods from ES5 to ES7.

File: es7-persistence/dependencies.lock [hunk 1]
```diff
@@ -14,6 +14,12 @@
         "com.google.guava:guava": {
             "locked": "30.0-jre"
         },
+        "com.google.inject:guice": {
```
Comment (saralaks) [inline es7-persistence/dependencies.lock:17]: These additions are not there in Netflix conductor version (https://github.com/Netflix/conductor/blob/main/es7-persistence/dependencies.lock).  Any reason to add these dependencies?  Can we sync with the latest netflix main to get their ES7 changes?

Comment (fthankac) [reply es7-persistence/dependencies.lock:17]: As mentioned in the description https://github.com/Netflix/conductor/tree/main/es7-persistence, for using es7 we need to delete all dependencies.lock files and then run:

./gradlew generateLock updateLock saveLock

The line com.google.inject:guice is generated during running above command.

**Comment by fthankac** [comment]
> changed these logs to debug.

Done

**Comment by fthankac** [comment]
> Please remove this log

Done.

**Comment by fthankac** [comment]
> Why these 2 new methods? Where we have defined those if it came as a part of conductor merge then lets remove our method and replace with these.

The two methods a) onWorkflowCompletedIfEnabled b) onWorkflowTerminatedIfEnabled are part of methods added in Netflix. 
By default  in the interface WorkflowStatusListener, implementation of these methods are below

default void onWorkflowCompletedIfEnabled(Workflow workflow) {
        if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
            onWorkflowCompleted(workflow);
        }
    }

    default void onWorkflowTerminatedIfEnabled(Workflow workflow) {
        if (workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
            onWorkflowTerminated(workflow);
        }
    }

Only if isWorkflowStatusListenerEnabled true then methods onWorkflowCompleted / onWorkflowTerminated are called.

Have overridden the menthods to directly call onWorkflowCompleted / onWorkflowTerminated instead of check.

**Comment by fthankac** [comment]
> Not sure where this method is called , please remove the call to onWorkflowTerminated and find where this is being called up?

Same comment as already specified.

**Review by saralaks — COMMENTED** [review]
remove es5-persistence folder as it's no longer valid

**Comment by fthankac** [comment]
> remove es5-persistence folder as it's no longer valid

Removed the es5-persistence folder.

**Review by saralaks — APPROVED** [review]

**Pull request merged**